packages/zoe/src/contracts/otcDeskReputation.js

```diff
@@ -0,0 +1,142 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { assert, details } from '@agoric/assert';
+import { trade } from '../contractSupport';
+
+import '../../exported';
+
+/**
+ * This OTC Desk contract does not assume that the underlying assets
+ * have been escrowed. Rather, a reputation mechanism
+ * is used to inform potential users of how often the quote is
+ * accurate, and how often the trade fails.
+ *
+ * @type {ContractStartFn}
+ */
+const start = zcf => {
+  const { zcfSeat: marketMakerSeat } = zcf.makeEmptySeatKit();
+
+  let numTradesAttempted = 0;
+  let numTradesDefected = 0;
+
+  const makeQuote = (price, assets, timerAuthority, deadline) => {
+    let quoteActive = true;
+    E(timerAuthority).setWakeup(
+      deadline,
+      harden({ wake: () => (quoteActive = false) }),
+    );
+    const acceptQuote = seat => {
+      assert(quoteActive, details`The quote was no longer active`);
+      const { want: userWantActual, give: userGiveActual } = seat.getProposal();
+      numTradesAttempted += 1;
+      const hasInventory = Object.entries(assets).every(([keyword, amount]) => {
+        const allocatedAmount = marketMakerSeat.getAmountAllocated(
+          keyword,
+          amount.brand,
+        );
+        return zcf.getAmountMath(amount.brand).isGTE(allocatedAmount, amount);
+      });
+      if (!hasInventory) {
+        numTradesDefected += 1;
+        throw Error(
```

Comment:
Does this exit the quote or could the user retry (and really slam the rating).

Reply:
This exits the quote. There's only one invitation per quote which cannot be used again